src/composables/useGetMessages.ts

```diff
@@ -106,76 +125,112 @@ export function useGetMessagesProvider() {
 	onBeforeUnmount(() => {
 		unsubscribe('networkOffline', handleNetworkOffline)
 		unsubscribe('networkOnline', handleNetworkOnline)
+		EventBus.off('route-change', onRouteChange)
 
-		store.dispatch('cancelPollNewMessages', { requestId: chatIdentifier.value })
+		store.dispatch('cancelPollNewMessages', { requestId: currentToken.value })
 		clearInterval(pollingTimeout)
 		clearInterval(expirationInterval)
 	})
 
+	/**
+	 * Parse hash string to get message id
+	 */
+	function getMessageIdFromHash(hash?: string): number | null {
+		return (hash && hash.startsWith('#message_')) ? parseInt(hash.slice(9), 10) : null
+	}
+
 	/**
 	 * Stop polling due to offline
 	 */
 	function handleNetworkOffline() {
-		console.debug('Canceling message request as we are offline')
-		store.dispatch('cancelPollNewMessages', { requestId: chatIdentifier.value })
+		if (currentToken.value) {
+			console.debug('Canceling message request as we are offline')
+			store.dispatch('cancelPollNewMessages', { requestId: currentToken.value })
+		}
 	}
 
 	/**
 	 * Resume polling, when back online
 	 */
 	function handleNetworkOnline() {
-		console.debug('Restarting polling of new chat messages')
-		pollNewMessages(currentToken.value)
+		if (currentToken.value) {
+			console.debug('Restarting polling of new chat messages')
+			pollNewMessages(currentToken.value)
+		}
+	}
+
+	/**
+	 * Handle route changes to initialize chat or thread, and focus given message
+	 */
+	async function onRouteChange({ from, to }: { from: RouteLocation, to: RouteLocation }) {
```

Comment:
```suggestion
	async function onRouteChange({ from, to }: { from: RouteLocation, to: RouteLocation }) {
		if (from.name !== to.name || from.name !== 'conversation' 
		|| from.params.token !== to.params.token || typeof to.params.token !== 'string') {
		return
	}
```

and remove the wrapper if